app.rb

```diff
@@ -0,0 +1,130 @@
+require './book'
+require './person'
+require './student'
+require './teacher'
+require './rental'
+
+class App
+  def initialize
+    @persons = []
+    @books = []
+    @rentals = []
+  end
+
+  def start_app
+    until choose_options
+      value = gets.chomp
+      if value == '7'
+        puts 'Thank You for patronage!'
+        break
+      end
+
+      chosen_value(value)
+    end
+  end
+
+  def list_all_books
+    puts 'There are No Books.' if @books.empty?
+    @books.each { |book| puts "Title: #{book.title} -> Author: #{book.author}" }
+  end
+
+  def list_all_people
+    puts 'There is no record of any person.' if @persons.empty?
+    @persons.each { |person| puts "Name: #{person.name} -> Age: #{person.age}, Id: #{person.id}" }
+  end
+
+  def create_person
+    puts 'To create a student, press 1, to create a teacher, press 2 : '
+    status = gets.chomp
+
+    case status
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'You can only choose 1 for student or 2 for teacher'
+    end
+  end
+
+  def create_student
+    puts 'Add Student'
+    puts 'Enter Student Age: '
+    age = gets.chomp.to_i
+    puts 'Enter Student Name: '
+    name = gets.chomp
+    puts 'Parent Permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+    case parent_permission
+    when 'n'
+      student = Student.new('undefined', age, name, parent_permission: false)
+      @persons << student
+      puts 'Student doesnt have parent permission, cant rent books'
+    when 'y'
+      student = Student.new('undefined', age, name, parent_permission: true)
+      @persons << student
+      puts 'One Student Added'
```

Comment:
- [ ] Hello and nice work but this code is returning an error while storing the student and teacher, the inputs are altered the name got the age and the age got 'undefined'.

![Capture d’écran 2022-08-31 122203](https://user-images.githubusercontent.com/61562626/187668108-4156a3aa-9ce7-412b-92cf-b445d796f3fd.png)